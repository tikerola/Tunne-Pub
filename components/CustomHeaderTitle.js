import React from 'react'
import { View, Text, StyleSheet } from 'react-native'

export default function CustomHeaderTitle({ name, letterNum, ...props }) {

  return (
    <View>
      <Text
        style={[props.style, { color: props.tintColor, fontSize: 20 }]}
      >
        TUNNE_APP <Text style={styles.name}>
          {
            name.length <= letterNum ? name : `${name.substring(0, letterNum)}...`
          }
        </Text></Text>
    </View>
  )
}

const styles = StyleSheet.create({

  name: {
    fontSize: 15,
    color: '#bbb'
  }
})
