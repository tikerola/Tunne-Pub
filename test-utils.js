import React from "react";
import { NavigationContext } from "@react-navigation/native";
import { render } from "@testing-library/react-native";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";
import CARD_DATA from "./data/cards";
import NEEDS_DATA from "./data/needs";
import { SENTENCES_DATA } from "./data/sentences";
import WORD_DATA from "./data/words";

const INITIAL_STATE = {
  cards: {
    // cards: CARD_DATA,
    // words: WORD_DATA,
    // needs: NEEDS_DATA,
    // sentences: SENTENCES_DATA,
    activeSummaryItemId: "123",
    summaryOfChosenItems: {
      123: {
        id: "123",
        name: "",
        cards: [],
        feelingsWords: [],
        feelingsYouMissWords: [],
        needs: ["hei"],
        needsBehindAFeeling: [],
        sentences: [],
        questions: {},
        itemSaved: true,
      },
    },
    screenNumber: 1,
    numOfItems: 0,
    initialLoadFinished: false,
  },
};

const mockStore = configureStore([]);
const store = mockStore(INITIAL_STATE);

const navContext = {
  isFocused: () => true,
  // addListener returns an unscubscribe function.
  addListener: jest.fn(() => jest.fn()),
};

const reduxRender = (ui, options, storeData = store) =>
  render(ui, {
    wrapper: ({ children }) => (
      <Provider store={storeData}>
        <NavigationContext.Provider value={navContext}>
          {children}
        </NavigationContext.Provider>
      </Provider>
    ),
    ...options,
  });

// re-export everything for convenience
export * from "@testing-library/react-native";

// override render method
export { reduxRender as render, store };
