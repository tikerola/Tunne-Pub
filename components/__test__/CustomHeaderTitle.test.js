import { render, rerender } from "../../test-utils";
import React from "react";
import CustomHeaderTitle from "../CustomHeaderTitle";

describe("<CustomHeaderTitle", () => {
  const name = "Holabaloo";
  const letterNum = 8;

  it("should match snapshot", () => {
    expect(
      render(<CustomHeaderTitle name={name} letterNum={letterNum} />)
    ).toMatchSnapshot();
  });

  it("should put three dots if name is longer than letternum letters", () => {
    let component = render(
      <CustomHeaderTitle name={name} letterNum={letterNum} />
    );
    expect(component.queryByText("Holabalo...")).toBeTruthy();

    component.rerender(
      <CustomHeaderTitle name={"Hola"} letterNum={letterNum} />
    );
    expect(component.queryByText("Hola")).toBeTruthy();
  });
});
