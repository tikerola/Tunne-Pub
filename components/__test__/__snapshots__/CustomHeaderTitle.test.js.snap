// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CustomHeaderTitle should match snapshot 1`] = `
<View>
  <Text
    style={
      Array [
        undefined,
        Object {
          "color": undefined,
          "fontSize": 20,
        },
      ]
    }
  >
    TUNNE_APP 
    <Text
      style={
        Object {
          "color": "#bbb",
          "fontSize": 15,
        }
      }
    >
      Holabalo...
    </Text>
  </Text>
</View>
`;
